### PR TITLE
Read build args for metricbeat system tests from file

### PR DESCRIPTION
**Pull request by jsoriano**
A new file is being added to add information about the supported
versions of a service, use this file to obtain the build args for
docker images needed by metricbeat system tests.

Python tests can now be decorated using the parameterized library
instead of manually copying them.

MySQL test files and docker compose definition has been moved
to its own directory.

More context about this change in https://github.com/elastic/beats/pull/14355#issuecomment-553987732.

### CODE REVIEW

File: metricbeat/module/mysql/supported-versions.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+variants:
+  - {'MYSQL_VARIANT': 'mysql',   'MYSQL_VERSION': '5.7.12'}
+  - {'MYSQL_VARIANT': 'mysql',   'MYSQL_VERSION': '8.0.13'}
+  - {'MYSQL_VARIANT': 'percona', 'MYSQL_VERSION': '5.7.24'}
+  - {'MYSQL_VARIANT': 'percona', 'MYSQL_VERSION': '8.0.13-4'}
+  - {'MYSQL_VARIANT': 'mariadb', 'MYSQL_VERSION': '10.2.23'}
+  - {'MYSQL_VARIANT': 'mariadb', 'MYSQL_VERSION': '10.3.14'}
+  - {'MYSQL_VARIANT': 'mariadb', 'MYSQL_VERSION': '10.4.4'}
```
Comment (jsoriano) [inline metricbeat/module/mysql/supported-versions.yml]: @mdelapenya when we agree on the format and location of this file I will open this for review.

Comment (jsoriano) [reply metricbeat/module/mysql/supported-versions.yml]: I have moved these files to the `_meta` directory of each module as in #14355.

File: CHANGELOG-developer.next.asciidoc [hunk 2]
```diff
@@ -60,3 +60,4 @@ The list below covers the major changes between 7.0.0-rc2 and master only.
 - Strip debug symbols from binaries to reduce binary sizes. {issue}12768[12768]
 - Compare event by event in `testadata` framework to avoid sorting problems {pull}13747[13747]
 - Added a `default_field` option to fields in fields.yml to offer a way to exclude fields from the default_field list. {issue}14262[14262] {pull}14341[14341]
+- `supported-versions.txt` can be used in metricbeat python system tests to obtain the build args for docker compose builds. {pull}14520[14520]
```
Comment (mdelapenya) [inline CHANGELOG-developer.next.asciidoc]: Typo? Isn't it `supported-versions.yml`? 🤔

File: metricbeat/module/mysql/_meta/supported-versions.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+variants:
+  - {'MYSQL_VARIANT': 'mysql',   'MYSQL_VERSION': '5.7.12'}
```
Comment (mdelapenya) [inline metricbeat/module/mysql/_meta/supported-versions.yml]: To be consistent across modules, I'd use this format:
```yaml
variants:
  - {'IMAGE': 'mysql',   'VERSION': '5.7.12'}
  - {'IMAGE': 'mysql',   'VERSION': '8.0.13'}
```
This way the parser is abstract

Comment (jsoriano) [reply metricbeat/module/mysql/_meta/supported-versions.yml]: Then we'll have to change all build args used at the moment, does it worth it?

To what parser do you refer?

Comment (mdelapenya) [reply metricbeat/module/mysql/_meta/supported-versions.yml]: Mmmm... you are right. The parser must use `MYSQL_VARIANT` (key of the map) as build-arg.

Thanks!

**Comment by jsoriano** [comment]
jenkins, test this again please

**Comment by jsoriano** [comment]
jenkins, test this again please

**Pull request merged**